<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Adjacency-Constrained Clustering of a Block-Diagonal Similarity Matrix • adjclust</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Adjacency-Constrained Clustering of a Block-Diagonal Similarity Matrix">
<meta name="description" content="Implements a constrained version of hierarchical agglomerative clustering, in which each observation is associated to a position, and only adjacent clusters can be merged. Typical application fields in bioinformatics include Genome-Wide Association Studies or Hi-C data analysis, where the similarity between items is a decreasing function of their genomic distance. Taking advantage of this feature, the implemented algorithm is time and memory efficient. This algorithm is described in Ambroise et al (2019) &lt;doi:10.1186/s13015-019-0157-4&gt;.">
<meta property="og:description" content="Implements a constrained version of hierarchical agglomerative clustering, in which each observation is associated to a position, and only adjacent clusters can be merged. Typical application fields in bioinformatics include Genome-Wide Association Studies or Hi-C data analysis, where the similarity between items is a decreasing function of their genomic distance. Taking advantage of this feature, the implemented algorithm is time and memory efficient. This algorithm is described in Ambroise et al (2019) &lt;doi:10.1186/s13015-019-0157-4&gt;.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">adjclust</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.6.10</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/hicClust.html">Clustering of Hi-C contact maps</a></li>
    <li><a class="dropdown-item" href="articles/notesCHAC.html">Implementation notes for the `adjclust` package</a></li>
    <li><a class="dropdown-item" href="articles/snpClust.html">Inferring Linkage Disequilibrium blocks from genotypes</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/pneuvial/adjclust/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="adjclust-adjacency-constrained-clustering">adjclust: Adjacency-constrained clustering<a class="anchor" aria-label="anchor" href="#adjclust-adjacency-constrained-clustering"></a>
</h1></div>
<p><a href="https://cran.r-project.org/package=adjclust" class="external-link"><img src="http://www.r-pkg.org/badges/version/adjclust" alt="CRAN_Status_Badge"></a> <a href="https://github.com/pneuvial/adjclust/actions" class="external-link"><img src="https://github.com/pneuvial/adjclust/workflows/R-CMD-check/badge.svg" alt="R build status"></a> <a href="https://app.codecov.io/github/pneuvial/adjclust/branch/develop" class="external-link"><img src="https://img.shields.io/codecov/c/github/pneuvial/adjclust/develop.svg" alt="Coverage Status"></a></p>
<p><code>adjclust</code> is a package that provides methods to perform adjacency-constrained hierarchical agglomerative clustering. Adjacency-constrained hierarchical agglomerative clustering is hierarchical agglomerative clustering (HAC) in which each observation is associated to a position, and the clustering is constrained so as only adjacent clusters are merged. It is useful in bioinformatics (e.g. Genome Wide Association Studies or Hi-C data analysis).</p>
<p><code>adjclust</code> provides three user level functions: <code>adjClust</code>, <code>snpClust</code> and <code>hicClust</code>, which are briefly explained below.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install adjclust from github with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"pneuvial/adjclust"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="adjclust">
<code>adjClust</code><a class="anchor" aria-label="anchor" href="#adjclust"></a>
</h2>
<p><code>adjClust</code> performs adjacency-constrained HAC for standard and sparse, similarity and dissimilarity matrices and <code>dist</code> objects. <code>Matrix::dgCMatrix</code> and <code>Matrix::dsCMatrix</code> are the supported sparse matrix classes. Let’s look at a basic example</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://pneuvial.github.io/adjclust/">"adjclust"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">sim</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1.0</span>, <span class="fl">0.5</span>, <span class="fl">0.2</span>, <span class="fl">0.1</span>,</span>
<span>                <span class="fl">0.5</span>, <span class="fl">1.0</span>, <span class="fl">0.1</span>, <span class="fl">0.2</span>,</span>
<span>                <span class="fl">0.2</span>, <span class="fl">0.1</span>, <span class="fl">1.0</span>, <span class="fl">0.6</span>,</span>
<span>                <span class="fl">0.1</span>, <span class="fl">0.2</span> ,<span class="fl">0.6</span> ,<span class="fl">1.0</span><span class="op">)</span>, nrow<span class="op">=</span><span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">h</span> <span class="op">&lt;-</span> <span class="fl">3</span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/adjClust.html">adjClust</a></span><span class="op">(</span><span class="va">sim</span>, <span class="st">"similarity"</span>, <span class="va">h</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-adjClust-1.png"><!-- --></p>
<p>The result is of class <code>chac</code>. It can be plotted as a dendrogram (as shown above). Successive merge and heights of clustering can be obtained by <code>fit$merge</code> and <code>fit$height</code> respectively.</p>
</div>
<div class="section level2">
<h2 id="snpclust">
<code>snpClust</code><a class="anchor" aria-label="anchor" href="#snpclust"></a>
</h2>
<p><code>snpClust</code> performs adjacency-constrained HAC for specific application of Genome Wide Association Studies (GWAS). A minimal example is given below. See <a href="vignettes/snpClust.Rmd">GWAS Vignette</a> for details.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"snpStats"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: survival</span></span>
<span><span class="co">#&gt; Loading required package: Matrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"ld.example"</span>, package <span class="op">=</span> <span class="st">"snpStats"</span><span class="op">)</span></span>
<span><span class="va">geno</span> <span class="op">&lt;-</span> <span class="va">ceph.1mb</span><span class="op">[</span>, <span class="op">-</span><span class="fl">316</span><span class="op">]</span>  <span class="co">## drop one SNP leading to one missing LD value</span></span>
<span><span class="va">h</span> <span class="op">&lt;-</span> <span class="fl">100</span></span>
<span><span class="va">ld.ceph</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/snpStats/man/ld.html" class="external-link">ld</a></span><span class="op">(</span><span class="va">geno</span>, stats <span class="op">=</span> <span class="st">"R.squared"</span>, depth <span class="op">=</span> <span class="va">h</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/image.html" class="external-link">image</a></span><span class="op">(</span><span class="va">ld.ceph</span>, lwd <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-snpClust-1.png"><!-- --></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/snpClust.html">snpClust</a></span><span class="op">(</span><span class="va">geno</span>, stats <span class="op">=</span> <span class="st">"R.squared"</span>, h <span class="op">=</span> <span class="va">h</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in run.snpClust(x, h = h, stats = stats): Forcing the LD similarity to</span></span>
<span><span class="co">#&gt; be smaller than or equal to 1</span></span>
<span><span class="co">#&gt; Note: 133 merges with non increasing heights.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: </span></span>
<span><span class="co">#&gt; Detected reversals in dendrogram: mode = 'corrected', 'within-disp' or 'total-disp' might be more relevant.</span></span></code></pre></div>
<p><img src="reference/figures/README-snpClust-2.png"><!-- --></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sel_clust</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/select.html">select</a></span><span class="op">(</span><span class="va">fit</span>, <span class="st">"bs"</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/plotSim.html">plotSim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">ld.ceph</span><span class="op">)</span>, clustering <span class="op">=</span> <span class="va">sel_clust</span>, dendro <span class="op">=</span> <span class="va">fit</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: </span></span>
<span><span class="co">#&gt; Detected reversals in dendrogram: mode = 'corrected', 'within-disp' or 'total-disp' might be more relevant.</span></span>
<span><span class="co">#&gt; Warning: Removed 602 rows containing missing values (`geom_text()`).</span></span></code></pre></div>
<p><img src="reference/figures/README-snpClust-3.png"><!-- --></p>
</div>
<div class="section level2">
<h2 id="hicclust">
<code>hicClust</code><a class="anchor" aria-label="anchor" href="#hicclust"></a>
</h2>
<p><code>hicClust</code> performs adjacency-constrained HAC for specific application of Hi-C data analysis. A minimal example is given below. See <a href="vignettes/hicClust.Rmd">Hi-C Vignette</a> for details.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"HiTC"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"hic_imr90_40_XX.rda"</span>, package <span class="op">=</span> <span class="st">"adjclust"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">binned</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/HiTC/man/binningC.html" class="external-link">binningC</a></span><span class="op">(</span><span class="va">hic_imr90_40_XX</span>, binsize <span class="op">=</span> <span class="fl">5e5</span><span class="op">)</span></span>
<span><span class="co">#&gt; Bin size 'xgi' =500488 [1x500488]</span></span>
<span><span class="co">#&gt; Bin size 'ygi' =500488 [1x500488]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/HiTC/man/mapC.html" class="external-link">mapC</a></span><span class="op">(</span><span class="va">binned</span><span class="op">)</span></span>
<span><span class="co">#&gt; minrange= 104  - maxrange= 36776.8</span></span></code></pre></div>
<p><img src="reference/figures/README-hicClust-1.png"><!-- --></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">fitB</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/hicClust.html">hicClust</a></span><span class="op">(</span><span class="va">binned</span><span class="op">)</span></span>
<span><span class="co">#&gt; Note: 5 merges with non increasing heights.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fitB</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: </span></span>
<span><span class="co">#&gt; Detected reversals in dendrogram: mode = 'corrected', 'within-disp' or 'total-disp' might be more relevant.</span></span></code></pre></div>
<p><img src="reference/figures/README-hicClust-2.png"><!-- --></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/plotSim.html">plotSim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/HiTC/man/HTCexp-class.html" class="external-link">intdata</a></span><span class="op">(</span><span class="va">binned</span><span class="op">)</span>, dendro <span class="op">=</span> <span class="va">fitB</span><span class="op">)</span> <span class="co"># default: log scale for colors</span></span>
<span><span class="co">#&gt; Warning: </span></span>
<span><span class="co">#&gt; Detected reversals in dendrogram: mode = 'corrected', 'within-disp' or 'total-disp' might be more relevant.</span></span>
<span><span class="co">#&gt; Warning: Removed 41 rows containing missing values (`geom_text()`).</span></span></code></pre></div>
<p><img src="reference/figures/README-hicClust-3.png"><!-- --></p>
</div>
<div class="section level2">
<h2 id="credits">Credits<a class="anchor" aria-label="anchor" href="#credits"></a>
</h2>
<p>Version 0.4.0 of this package was completed by Shubham Chaturvedi as a part of the <a href="https://summerofcode.withgoogle.com/projects/#4961904920363008" class="external-link">Google Summer of Code 2017</a> program.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=adjclust" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/pneuvial/adjclust/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/pneuvial/adjclust/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing adjclust</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="http://www.math-evry.cnrs.fr/members/cambroise/" class="external-link">Christophe Ambroise</a> <br><small class="roles"> Author </small>  </li>
<li>Shubham Chaturvedi <br><small class="roles"> Author </small>  </li>
<li>Alia Dehman <br><small class="roles"> Author </small>  </li>
<li>
<a href="https://www.math.univ-toulouse.fr/~pneuvial/" class="external-link">Pierre Neuvial</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-3584-9998" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Guillem Rigaill <br><small class="roles"> Author </small>  </li>
<li>
<a href="http://www.nathalievialaneix.eu" class="external-link">Nathalie Vialaneix</a> <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-1156-0639" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Gabriel Hoffman <br><small class="roles"> Author </small>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="http://www.math-evry.cnrs.fr/members/cambroise/" class="external-link">Christophe Ambroise</a>, Shubham Chaturvedi, Alia Dehman, <a href="https://www.math.univ-toulouse.fr/~pneuvial/" class="external-link">Pierre Neuvial</a>, Guillem Rigaill, <a href="http://www.nathalievialaneix.eu" class="external-link">Nathalie Vialaneix</a>, Gabriel Hoffman.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
