---
title: "Adjacency-constrained hierarchical clustering of Single Nucleotide Polymorphisms"
author: "Pierre Neuvial, Shubham Chaturvedi"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Adjacency-constrained hierarchical clustering of Single Nucleotide Polymorphisms}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

In this vignette we demonstrate the use of `snpClust` function in the `adjclust` package. `snpClust` performs adjacency-constrained hierarchical clustering of single nucleotide polymorphisms (SNPs), where the similarity between SNPs is defined by linkage disequilibrium (LD). 

This function implements the algorithm described in the third chapter of [2]. It is an extension of the algorithm described in [3]. Denoting by $p$ the number of SNPs to cluster and assuming that the similarity between SNPs whose indices are more distant than $h$, its time complexity is $O(p (\log(p) + h))$, and its space complexity is $O(hp)$.

```{r, message = FALSE}
library("adjclust")
```

## Loading and displaying genotype data

The begining of this vignette closely follows the "LD vignette" of the SnpStats package [1]. First, we load genotype data.

```{r, results="hide", message=FALSE}
library("matrixStats")
library("snpStats")
data("ld.example", package="snpStats")
```

```{r, echo=FALSE}
p <- ncol(ceph.1mb)
nSamples <- nrow(ceph.1mb)
```

We focus on the `ceph.1mb` data. These data are drawn from the International HapMap Project and concern `r p` SNPs over a 1Mb region of chromosome 22 in sample of `r nSamples` Europeans.

```{r}
ceph.1mb
```

We can compute and display the LD between these SNPs.

```{r}
ld.ceph <- ld(ceph.1mb, stats="R.squared", depth=p-1)
image(ld.ceph, lwd=0)
```

## Adjacency-constrained Hierarchical Agglomerative Clustering

The above figure suggests that the LD signal is concentrated close to the diagonal. Therefore, we focus on a diagonal band:

```{r}
h <- 100
ld.ceph <- ld(ceph.1mb, stats="R.squared", depth=h)
image(ld.ceph, lwd=0)
```

In this unrealistically small example with only 90 subjects, it turns out that one of the LD estimates is NA due to the lack of genetic diversity in the sample:

```{r}
any(is.na(ld.ceph))
which(is.na(ld.ceph), arr.ind = TRUE)
```

As a quick fix for this vignette, we modify one of the observed genotypes as follows, and recalculate the LD:

```{r}
ceph.1mb[4, 286]@.Data[1,1] <- as.raw(3) 
ld.ceph <- ld(ceph.1mb, stats="R.squared", depth=h)  
```

Due to numerical errors in the LD estimation, some of the estimated values are slightly larger than 1, so we round the estimates to 1e-10:

```{r}
max(ld.ceph-1)
ld.ceph <- round(ld.ceph, digits=10)
```


## Using snpClust function with different input classes

###Case 1: Input belongs to class Matrix::dgCMatrix generated by snpStats::ld function
```{r}
class(ld.ceph)
fit1 <- snpClust(ld.ceph, h)
```

###Case 2: Input belongs to class snpStats::SnpMatrix

```{r}
class(ceph.1mb)
fit2 <- snpClust(ceph.1mb, h, "R.squared")  
```

###Case3: Input belongs class base::matrix

```{r, warning = FALSE}
ceph.1mb <- as(ceph.1mb, "matrix")
class(ceph.1mb)
fit3 <- snpClust(ceph.1mb, h, "R.squared")  
```
  
## Output

The output of the `snpClust` is of class "hclust". In particular, it can be plotted as a dendrogram:

```{r}
plot(fit1)
```


Moreover, the output contains an element named `merge` which describes the successive merges of the clustering, and an element `gains` which gives the improvement in the criterion optimized by the clustering at each successive merge.

```{r}
head(cbind(fit1$merge, fit1$gains))
```



For equivalent input in different classes, we get equivalent results.In this example, fit1, fit2 and fit3 are equivalent.

```{r}
all.equal(fit1$merge, fit2$merge)
all.equal(fit1$gains, fit2$gains)

all.equal(fit2$merge, fit3$merge)
all.equal(fit2$gains, fit3$gains)
```

## References

[1] Clayton, D. (2015). snpStats: SnpMatrix and XSnpMatrix classes and methods. R package version 1.20.0

[2] Dehman A. (2015). [Spatial clustering of linkage disequilibrium blocks for genome-wide association studies](https://tel.archives-ouvertes.fr/tel-01288568/). Phd Thesis, UniversitÃ© Paris Saclay.

[3] Dehman, A. Ambroise, C. and Neuvial, P. (2015). Performance of a blockwise approach in variable selection using linkage disequilibrium information. *BMC Bioinformatics* 16:148.

## Session information

```{r}
sessionInfo()
```
