<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Clustering of Hi-C contact maps • adjclust</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Clustering of Hi-C contact maps">
<meta property="og:description" content="adjclust">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">adjclust</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.5.99</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/hicClust.html">Clustering of Hi-C contact maps</a>
    </li>
    <li>
      <a href="../articles/notesCHAC.html">Implementation notes for the `adjclust` package</a>
    </li>
    <li>
      <a href="../articles/snpClust.html">Inferring Linkage Disequilibrium blocks from genotypes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/pneuvial/adjclust/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Clustering of Hi-C contact maps</h1>
                        <h4 class="author">Shubham Chaturvedi, Pierre Neuvial, Nathalie Villa-Vialaneix</h4>
            
            <h4 class="date">2020-06-15</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pneuvial/adjclust/blob/master/vignettes/hicClust.Rmd"><code>vignettes/hicClust.Rmd</code></a></small>
      <div class="hidden name"><code>hicClust.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="co"># this vignette is not created if HiTC is not installed</span>
<span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/library.html">require</a></span>(<span class="st">"HiTC"</span>, <span class="kw">quietly</span> <span class="kw">=</span> <span class="fl">TRUE</span>)) {
  <span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="no"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html">opts_chunk</a></span>$<span class="fu">set</span>(<span class="kw">eval</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
}</pre></body></html></div>
<pre><code>## 
## Attaching package: 'BiocGenerics'</code></pre>
<pre><code>## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which, which.max, which.min</code></pre>
<pre><code>## 
## Attaching package: 'S4Vectors'</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     expand.grid</code></pre>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Hi-C is a sequencing-based molecular assay designed to measure intra and interchromosomal interactions between the DNA molecule. In particular, the identification of Topologically-Associated Domains (TADs), that is, of regions of the genome in which physical interactions are frequent, provides insight into the three-dimensional organization of a genome [1].</p>
<p>Hi-C data are in the form of two-dimensional <em>contact maps</em>, <em>i.e.</em>, matrices whose <span class="math inline">\(i,j\)</span> entry quantifies the intensity of the physical interaction between two genome regions <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> at the DNA level. In this vignette, we demonstrate the use of <code><a href="../reference/hicClust.html">adjclust::hicClust</a></code> to perform adjacency-constrained hierarchical agglomerative clustering (HAC) of Hi-C contact maps. The output of this function is a dendrogram, which can be cut to identify TADs. The algorithm used for adjacency-constrained (HAC) is described in the third chapter of [2].</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"adjclust"</span>)</pre></body></html></div>
</div>
<div id="loading-and-displaying-a-sample-hi-c-contact-map" class="section level1">
<h1 class="hasAnchor">
<a href="#loading-and-displaying-a-sample-hi-c-contact-map" class="anchor"></a>Loading and displaying a sample Hi-C contact map</h1>
<p>The data set <code>hic_imr90_40_XX</code> is an object of class <code>HTCexp</code> which has been obtained from the <code>HiTC</code> package [3]. It is a contact map corresponding to the first 500 x 500 bins on chromosome X vs chromosome X.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"hic_imr90_40_XX.rda"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"adjclust"</span>))</pre></body></html></div>
<p>The script used to create this map can be found by executing the following command:</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"system/create_hic_chrXchrX.R"</span>, <span class="kw">package</span><span class="kw">=</span><span class="st">"adjclust"</span>)</pre></body></html></div>
<p>Now we have a look at the data.</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="kw pkg">HiTC</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiTC/man/mapC.html">mapC</a></span>(<span class="no">hic_imr90_40_XX</span>)</pre></body></html></div>
<p><img src="hicClust_files/figure-html/mapHiC-1.png" width="700"></p>
</div>
<div id="using-hicclust" class="section level1">
<h1 class="hasAnchor">
<a href="#using-hicclust" class="anchor"></a>Using <code>hicClust</code>
</h1>
<p><code>hicClust</code> operates directly on objects of class <code>HTCexp</code></p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="no">fit</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/hicClust.html">hicClust</a></span>(<span class="no">hic_imr90_40_XX</span>)</pre></body></html></div>
<pre><code>## Note: 137 merges with non increasing heights.</code></pre>
<p>It is also possible to work on binned data. Below we choose a bin size of <span class="math inline">\(5 \times 10^5\)</span>:</p>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="no">binned</span> <span class="kw">&lt;-</span> <span class="kw pkg">HiTC</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiTC/man/binningC.html">binningC</a></span>(<span class="no">hic_imr90_40_XX</span>, <span class="kw">binsize</span> <span class="kw">=</span> <span class="fl">1e5</span>)
<span class="kw pkg">HiTC</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiTC/man/mapC.html">mapC</a></span>(<span class="no">binned</span>)</pre></body></html></div>
<p><img src="hicClust_files/figure-html/binning-1.png" width="700"></p>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="no">fitB</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/hicClust.html">hicClust</a></span>(<span class="no">binned</span>)
<span class="no">fitB</span></pre></body></html></div>
<pre><code>## 
## Call:
## run.adjclust(mat = mat, type = type, h = h)
## 
## Cluster method   : hicClust 
## Number of objects: 205</code></pre>
<p>The output is of class <code>chac</code>. In particular, it can be plotted as a dendrogram silently relying on the function <code>plot.dendrogram</code>:</p>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span>(<span class="no">fitB</span>)</pre></body></html></div>
<pre><code>## Warning in plot.chac(fitB): 
## Detected reversals in dendrogram: mode = 'corrected', 'within-disp' or 'total-disp' might be more relevant.</code></pre>
<p><img src="hicClust_files/figure-html/dendro-1.png" width="700"></p>
<p>Moreover, the output contains an element named <code>merge</code> which describes the successive merges of the clustering, and an element <code>gains</code> which gives the improvement in the criterion optimized by the clustering at each successive merge.</p>
<div class="sourceCode" id="cb19"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span>(<span class="no">fitB</span>$<span class="no">merge</span>, <span class="no">fitB</span>$<span class="no">gains</span>))</pre></body></html></div>
<pre><code>##      [,1] [,2]
## [1,]   -3   -4
## [2,]   -2    1
## [3,]    2   -5
## [4,]   -1    3
## [5,]    4   -6
## [6,]  -17  -18</code></pre>
</div>
<div id="other-types-of-input" class="section level1">
<h1 class="hasAnchor">
<a href="#other-types-of-input" class="anchor"></a>Other types of input</h1>
<p>Contacts maps can also be stored as objects of class <code>Matrix::dsCMatrix</code>, or as plain text files. These types of input are also accepted as first argument to <code>hicClust</code>.</p>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<p>[1] Dixon J.R., <em>et al</em> (2012). Topological domains in mammalian genomes identified by analysis of chromatin interactions. <em>Nature</em>, <strong>485</strong>(7398), 376.</p>
<p>[2] Dehman A. (2015). <a href="https://tel.archives-ouvertes.fr/tel-01288568/">Spatial clustering of linkage disequilibrium blocks for genome-wide association studies</a>. Phd Thesis, Université Paris Saclay.</p>
<p>[3] Servant N., <em>et al</em> (2012). HiTC: Exploration of High-Throughput ‘C’ experiments. <em>Bioinformatics</em>, <strong>28</strong>(21), 2843-2844.</p>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<div class="sourceCode" id="cb21"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</pre></body></html></div>
<pre><code>## R version 4.0.0 alpha (2020-04-05 r78150)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Mojave 10.14.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/fr_FR.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
## [1] adjclust_0.5.99      HiTC_1.32.0          GenomicRanges_1.40.0
## [4] GenomeInfoDb_1.24.0  IRanges_2.22.2       S4Vectors_0.26.1    
## [7] BiocGenerics_0.34.0 
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.4                  compiler_4.0.0             
##  [3] RColorBrewer_1.1-2          XVector_0.28.0             
##  [5] bitops_1.0-6                tools_4.0.0                
##  [7] zlibbioc_1.34.0             digest_0.6.25              
##  [9] evaluate_0.14               memoise_1.1.0              
## [11] lattice_0.20-41             rlang_0.4.5                
## [13] Matrix_1.2-18               DelayedArray_0.14.0        
## [15] rstudioapi_0.11             yaml_2.2.1                 
## [17] pkgdown_1.5.1               xfun_0.12                  
## [19] GenomeInfoDbData_1.2.3      rtracklayer_1.47.0         
## [21] stringr_1.4.0               knitr_1.28                 
## [23] desc_1.2.0                  Biostrings_2.56.0          
## [25] fs_1.3.2                    rprojroot_1.3-2            
## [27] grid_4.0.0                  Biobase_2.48.0             
## [29] R6_2.4.1                    BiocParallel_1.22.0        
## [31] XML_3.99-0.3                rmarkdown_2.1              
## [33] magrittr_1.5                matrixStats_0.56.0         
## [35] GenomicAlignments_1.24.0    Rsamtools_2.4.0            
## [37] backports_1.1.5             htmltools_0.4.0            
## [39] MASS_7.3-51.5               SummarizedExperiment_1.18.1
## [41] assertthat_0.2.1            capushe_1.1.1              
## [43] stringi_1.4.6               RCurl_1.98-1.2             
## [45] crayon_1.3.4</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="http://www.math-evry.cnrs.fr/members/cambroise/">Christophe Ambroise</a>, Shubham Chaturvedi, Alia Dehman, <a href="https://www.math.univ-toulouse.fr/~pneuvial/">Pierre Neuvial</a>, Guillem Rigaill, <a href="http://www.nathalievialaneix.eu">Nathalie Vialaneix</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
