<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Clustering of Hi-C contact maps • adjclust</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Clustering of Hi-C contact maps">
<meta property="og:description" content="adjclust">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">adjclust</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.6.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/hicClust.html">Clustering of Hi-C contact maps</a>
    </li>
    <li>
      <a href="../articles/notesCHAC.html">Implementation notes for the `adjclust` package</a>
    </li>
    <li>
      <a href="../articles/snpClust.html">Inferring Linkage Disequilibrium blocks from genotypes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/pneuvial/adjclust/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Clustering of Hi-C contact maps</h1>
                        <h4 data-toc-skip class="author">Shubham Chaturvedi, Pierre Neuvial, Nathalie Vialaneix</h4>
            
            <h4 data-toc-skip class="date">2022-03-31</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pneuvial/adjclust/blob/HEAD/vignettes/hicClust.Rmd" class="external-link"><code>vignettes/hicClust.Rmd</code></a></small>
      <div class="hidden name"><code>hicClust.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># IMPORTANT: this vignette can not be created if HiTC is not installed</span>
<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"HiTC"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>eval <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">## Attaching package: 'BiocGenerics'</span></code></pre>
<pre><code><span class="co">## The following objects are masked from 'package:stats':</span>
<span class="co">## </span>
<span class="co">##     IQR, mad, sd, var, xtabs</span></code></pre>
<pre><code><span class="co">## The following objects are masked from 'package:base':</span>
<span class="co">## </span>
<span class="co">##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
<span class="co">##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
<span class="co">##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
<span class="co">##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
<span class="co">##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
<span class="co">##     union, unique, unsplit, which.max, which.min</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## Attaching package: 'S4Vectors'</span></code></pre>
<pre><code><span class="co">## The following objects are masked from 'package:base':</span>
<span class="co">## </span>
<span class="co">##     expand.grid, I, unname</span></code></pre>
<div class="section level1">
<h1 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h1>
<p>Hi-C is a sequencing-based molecular assay designed to measure intra and inter-chromosomal interactions between the DNA molecule. In particular, the identification of Topologically-Associated Domains (TADs), that is, of regions of the genome in which physical interactions are frequent, provides insight into the three-dimensional organization of a genome [2].</p>
<p>Hi-C data are in the form of two-dimensional <em>contact maps</em>, <em>i.e.</em>, matrices whose <span class="math inline">\(i,j\)</span> entry quantifies the intensity of the physical interaction between two genome regions <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> at the DNA level. In this vignette, we demonstrate the use of <code><a href="../reference/hicClust.html">adjclust::hicClust</a></code> to perform adjacency-constrained hierarchical agglomerative clustering (HAC) of Hi-C contact maps. The output of this function is a dendrogram, which can be cut to identify TADs. The algorithm used for adjacency-constrained (HAC) is described in [3,4].</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://pneuvial.github.io/adjclust/">"adjclust"</a></span><span class="op">)</span></code></pre></div>
</div>
<div class="section level1">
<h1 id="loading-and-displaying-a-sample-hi-c-contact-map">Loading and displaying a sample Hi-C contact map<a class="anchor" aria-label="anchor" href="#loading-and-displaying-a-sample-hi-c-contact-map"></a>
</h1>
<p>The data set <code>hic_imr90_40_XX</code> is an object of class <code>HTCexp</code> which has been obtained from the <code>HiTC</code> package [4]. It is a contact map corresponding to the first 500 x 500 bins on chromosome X vs chromosome X.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"hic_imr90_40_XX.rda"</span>, package <span class="op">=</span> <span class="st">"adjclust"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>The script used to create this map can be found by executing the following command:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"system/create_hic_chrXchrX.R"</span>, package<span class="op">=</span><span class="st">"adjclust"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "/home/nathalie/R/x86_64-pc-linux-gnu-library/4.1/adjclust/system/create_hic_chrXchrX.R"</span></code></pre>
<p>Now we have a look at the data.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">HiTC</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiTC/man/mapC.html" class="external-link">mapC</a></span><span class="op">(</span><span class="va">hic_imr90_40_XX</span><span class="op">)</span></code></pre></div>
<p><img src="hicClust_files/figure-html/mapHiC-1.png" width="700"></p>
</div>
<div class="section level1">
<h1 id="using-hicclust">Using <code>hicClust</code><a class="anchor" aria-label="anchor" href="#using-hicclust"></a>
</h1>
<p><code>hicClust</code> operates directly on objects of class <code>HTCexp</code></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hicClust.html">hicClust</a></span><span class="op">(</span><span class="va">hic_imr90_40_XX</span><span class="op">)</span></code></pre></div>
<p>It is also possible to work on binned data. Below we choose a bin size of <span class="math inline">\(5 \times 10^5\)</span>:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">binned</span> <span class="op">&lt;-</span> <span class="fu">HiTC</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiTC/man/binningC.html" class="external-link">binningC</a></span><span class="op">(</span><span class="va">hic_imr90_40_XX</span>, binsize <span class="op">=</span> <span class="fl">1e5</span><span class="op">)</span>
<span class="va">fitB</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hicClust.html">hicClust</a></span><span class="op">(</span><span class="va">binned</span><span class="op">)</span>
<span class="va">fitB</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">## Call:</span>
<span class="co">## run.adjclust(mat = mat, type = type, h = h, strictCheck = strictCheck)</span>
<span class="co">## </span>
<span class="co">## Cluster method   : hicClust </span>
<span class="co">## Number of objects: 205</span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">HiTC</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiTC/man/mapC.html" class="external-link">mapC</a></span><span class="op">(</span><span class="va">binned</span><span class="op">)</span></code></pre></div>
<p><img src="hicClust_files/figure-html/plotBinned-1.png" width="700"></p>
<p>The output is of class <code>chac</code>. In particular, it can be plotted as a dendrogram silently relying on the function <code>plot.dendrogram</code>:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fitB</span>, mode <span class="op">=</span> <span class="st">"corrected"</span><span class="op">)</span></code></pre></div>
<p><img src="hicClust_files/figure-html/dendro-1.png" width="700"></p>
<p>Moreover, the output contains an element named <code>merge</code> which describes the successive merges of the clustering, and an element <code>gains</code> which gives the improvement in the criterion optimized by the clustering at each successive merge.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">fitB</span><span class="op">$</span><span class="va">merge</span>, <span class="va">fitB</span><span class="op">$</span><span class="va">gains</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##      [,1] [,2]</span>
<span class="co">## [1,]   -3   -4</span>
<span class="co">## [2,]   -2    1</span>
<span class="co">## [3,]    2   -5</span>
<span class="co">## [4,]   -1    3</span>
<span class="co">## [5,]    4   -6</span>
<span class="co">## [6,]  -17  -18</span></code></pre>
</div>
<div class="section level1">
<h1 id="other-types-of-input">Other types of input<a class="anchor" aria-label="anchor" href="#other-types-of-input"></a>
</h1>
<p>Contacts maps can also be stored as objects of class <code>Matrix::dsCMatrix</code>, or as plain text files. These types of input are also accepted as first argument to <code>hicClust</code>.</p>
</div>
<div class="section level1">
<h1 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h1>
<p>[1] Ambroise C., Dehman A., Neuvial P., Rigaill G., and Vialaneix N. (2019). Adjacency-constrained hierarchical clustering of a band similarity matrix with application to genomics. <em>Algorithms for Molecular Biology</em>, <strong>14</strong>, 22.</p>
<p>[2] Dixon J.R., <em>et al</em> (2012). Topological domains in mammalian genomes identified by analysis of chromatin interactions. <em>Nature</em>, <strong>485</strong>(7398), 376.</p>
<p>[3] Randriamihamison N., Vialaneix N., and Neuvial P. (2021). Applicability and interpretability of Ward’s hierarchical agglomerative clustering with or without contiguity constraints. <em>Journal of Classification</em>, <strong>38</strong>, 363–389.</p>
<p>[4] Servant N., <em>et al</em> (2012). HiTC: Exploration of High-Throughput ‘C’ experiments. <em>Bioinformatics</em>, <strong>28</strong>(21), 2843-2844.</p>
</div>
<div class="section level1">
<h1 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h1>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.2 (2021-11-01)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 20.04.3 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=fr_FR.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=fr_FR.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=fr_FR.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=fr_FR.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">## [1] adjclust_0.6.4       HiTC_1.38.0          GenomicRanges_1.46.1</span>
<span class="co">## [4] GenomeInfoDb_1.30.0  IRanges_2.28.0       S4Vectors_0.32.3    </span>
<span class="co">## [7] BiocGenerics_0.40.0 </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##  [1] SummarizedExperiment_1.24.0 capushe_1.1.1              </span>
<span class="co">##  [3] xfun_0.24                   bslib_0.3.1                </span>
<span class="co">##  [5] purrr_0.3.4                 lattice_0.20-45            </span>
<span class="co">##  [7] htmltools_0.5.2             rtracklayer_1.54.0         </span>
<span class="co">##  [9] yaml_2.2.1                  XML_3.99-0.9               </span>
<span class="co">## [11] rlang_0.4.11                pkgdown_2.0.1              </span>
<span class="co">## [13] jquerylib_0.1.4             BiocParallel_1.28.3        </span>
<span class="co">## [15] RColorBrewer_1.1-2          matrixStats_0.61.0         </span>
<span class="co">## [17] GenomeInfoDbData_1.2.7      stringr_1.4.0              </span>
<span class="co">## [19] zlibbioc_1.40.0             MatrixGenerics_1.6.0       </span>
<span class="co">## [21] Biostrings_2.62.0           ragg_1.2.1                 </span>
<span class="co">## [23] memoise_2.0.1               evaluate_0.14              </span>
<span class="co">## [25] restfulr_0.0.13             Biobase_2.54.0             </span>
<span class="co">## [27] knitr_1.33                  fastmap_1.1.0              </span>
<span class="co">## [29] parallel_4.1.2              highr_0.9                  </span>
<span class="co">## [31] Rcpp_1.0.7                  cachem_1.0.6               </span>
<span class="co">## [33] DelayedArray_0.20.0         desc_1.4.0                 </span>
<span class="co">## [35] jsonlite_1.7.2              XVector_0.34.0             </span>
<span class="co">## [37] systemfonts_1.0.3           fs_1.5.0                   </span>
<span class="co">## [39] Rsamtools_2.10.0            textshaping_0.3.6          </span>
<span class="co">## [41] rjson_0.2.21                digest_0.6.27              </span>
<span class="co">## [43] stringi_1.6.2               BiocIO_1.4.0               </span>
<span class="co">## [45] rprojroot_2.0.2             grid_4.1.2                 </span>
<span class="co">## [47] tools_4.1.2                 bitops_1.0-7               </span>
<span class="co">## [49] magrittr_2.0.1              sass_0.4.0                 </span>
<span class="co">## [51] RCurl_1.98-1.5              crayon_1.4.1               </span>
<span class="co">## [53] MASS_7.3-55                 Matrix_1.4-1               </span>
<span class="co">## [55] sparseMatrixStats_1.6.0     rmarkdown_2.10             </span>
<span class="co">## [57] R6_2.5.1                    GenomicAlignments_1.30.0   </span>
<span class="co">## [59] compiler_4.1.2</span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="http://www.math-evry.cnrs.fr/members/cambroise/" class="external-link">Christophe Ambroise</a>, Shubham Chaturvedi, Alia Dehman, <a href="https://www.math.univ-toulouse.fr/~pneuvial/" class="external-link">Pierre Neuvial</a>, Guillem Rigaill, <a href="http://www.nathalievialaneix.eu" class="external-link">Nathalie Vialaneix</a>, Gabriel Hoffman.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
