<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Clustering of Hi-C contact maps â€¢ adjclust</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Clustering of Hi-C contact maps">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">adjclust</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.6.10</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/hicClust.html">Clustering of Hi-C contact maps</a></li>
    <li><a class="dropdown-item" href="../articles/notesCHAC.html">Implementation notes for the `adjclust` package</a></li>
    <li><a class="dropdown-item" href="../articles/snpClust.html">Inferring Linkage Disequilibrium blocks from genotypes</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/pneuvial/adjclust/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Clustering of Hi-C contact maps</h1>
                        <h4 data-toc-skip class="author">Shubham
Chaturvedi, Pierre Neuvial, Nathalie Vialaneix</h4>
            
            <h4 data-toc-skip class="date">2024-10-08</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pneuvial/adjclust/blob/master/vignettes/hicClust.Rmd" class="external-link"><code>vignettes/hicClust.Rmd</code></a></small>
      <div class="d-none name"><code>hicClust.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># IMPORTANT: this vignette can not be created if HiTC is not installed</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"HiTC"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>eval <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">##     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">##     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">##     Position, rank, rbind, Reduce, rownames, sapply, setdiff, table,</span></span>
<span><span class="co">##     tapply, union, unique, unsplit, which.max, which.min</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'S4Vectors'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:utils':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     findMatches</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     expand.grid, I, unname</span></span></code></pre>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Hi-C is a sequencing-based molecular assay designed to measure intra
and inter-chromosomal interactions between the DNA molecule. In
particular, the identification of Topologically-Associated Domains
(TADs), that is, of regions of the genome in which physical interactions
are frequent, provides insight into the three-dimensional organization
of a genome [2].</p>
<p>Hi-C data are in the form of two-dimensional <em>contact maps</em>,
<em>i.e.</em>, matrices whose
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>i</mi><mo>,</mo><mi>j</mi></mrow><annotation encoding="application/x-tex">i,j</annotation></semantics></math>
entry quantifies the intensity of the physical interaction between two
genome regions
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>i</mi><annotation encoding="application/x-tex">i</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>j</mi><annotation encoding="application/x-tex">j</annotation></semantics></math>
at the DNA level. In this vignette, we demonstrate the use of
<code><a href="../reference/hicClust.html">adjclust::hicClust</a></code> to perform adjacency-constrained
hierarchical agglomerative clustering (HAC) of Hi-C contact maps. The
output of this function is a dendrogram, which can be cut to identify
TADs. The algorithm used for adjacency-constrained (HAC) is described in
[3,4].</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://pneuvial.github.io/adjclust/">"adjclust"</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="loading-and-displaying-a-sample-hi-c-contact-map">Loading and displaying a sample Hi-C contact map<a class="anchor" aria-label="anchor" href="#loading-and-displaying-a-sample-hi-c-contact-map"></a>
</h2>
<p>The data set <code>hic_imr90_40_XX</code> is an object of class
<code>HTCexp</code> which has been obtained from the <code>HiTC</code>
package [4]. It is a contact map corresponding to the first 500 x 500
bins on chromosome X vs chromosome X.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"hic_imr90_40_XX.rda"</span>, package <span class="op">=</span> <span class="st">"adjclust"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>The script used to create this map can be found by executing the
following command:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"system/create_hic_chrXchrX.R"</span>, package<span class="op">=</span><span class="st">"adjclust"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "/home/runner/work/_temp/Library/adjclust/system/create_hic_chrXchrX.R"</span></span></code></pre>
<p>Now we have a look at the data.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">HiTC</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiTC/man/mapC.html" class="external-link">mapC</a></span><span class="op">(</span><span class="va">hic_imr90_40_XX</span><span class="op">)</span></span></code></pre></div>
<p><img src="hicClust_files/figure-html/mapHiC-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="using-hicclust">Using <code>hicClust</code><a class="anchor" aria-label="anchor" href="#using-hicclust"></a>
</h2>
<p><code>hicClust</code> operates directly on objects of class
<code>HTCexp</code></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hicClust.html">hicClust</a></span><span class="op">(</span><span class="va">hic_imr90_40_XX</span><span class="op">)</span></span></code></pre></div>
<p>It is also possible to work on binned data. Below we choose a bin
size of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>5</mn><mo>Ã—</mo><msup><mn>10</mn><mn>5</mn></msup></mrow><annotation encoding="application/x-tex">5 \times 10^5</annotation></semantics></math>:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">binned</span> <span class="op">&lt;-</span> <span class="fu">HiTC</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiTC/man/binningC.html" class="external-link">binningC</a></span><span class="op">(</span><span class="va">hic_imr90_40_XX</span>, binsize <span class="op">=</span> <span class="fl">1e5</span><span class="op">)</span></span>
<span><span class="va">fitB</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hicClust.html">hicClust</a></span><span class="op">(</span><span class="va">binned</span><span class="op">)</span></span>
<span><span class="va">fitB</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Call:</span></span>
<span><span class="co">## hicClust(binned)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Cluster method   : hicClust </span></span>
<span><span class="co">## Number of objects: 205</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">HiTC</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiTC/man/mapC.html" class="external-link">mapC</a></span><span class="op">(</span><span class="va">binned</span><span class="op">)</span></span></code></pre></div>
<p><img src="hicClust_files/figure-html/plotBinned-1.png" width="700"></p>
<p>The output is of class <code>chac</code>. In particular, it can be
plotted as a dendrogram silently relying on the function
<code>plot.dendrogram</code>:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fitB</span>, mode <span class="op">=</span> <span class="st">"corrected"</span><span class="op">)</span></span></code></pre></div>
<p><img src="hicClust_files/figure-html/dendro-1.png" width="700"></p>
<p>Moreover, the output contains an element named <code>merge</code>
which describes the successive merges of the clustering, and an element
<code>gains</code> which gives the improvement in the criterion
optimized by the clustering at each successive merge.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">fitB</span><span class="op">$</span><span class="va">merge</span>, <span class="va">fitB</span><span class="op">$</span><span class="va">gains</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##      [,1] [,2]</span></span>
<span><span class="co">## [1,]   -3   -4</span></span>
<span><span class="co">## [2,]   -2    1</span></span>
<span><span class="co">## [3,]    2   -5</span></span>
<span><span class="co">## [4,]   -1    3</span></span>
<span><span class="co">## [5,]    4   -6</span></span>
<span><span class="co">## [6,]  -17  -18</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="other-types-of-input">Other types of input<a class="anchor" aria-label="anchor" href="#other-types-of-input"></a>
</h2>
<p>Contacts maps can also be stored as objects of class
<code>Matrix::dsCMatrix</code>, or as plain text files. These types of
input are also accepted as first argument to <code>hicClust</code>.</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>[1] Ambroise C., Dehman A., Neuvial P., Rigaill G., and Vialaneix N.
(2019). Adjacency-constrained hierarchical clustering of a band
similarity matrix with application to genomics. <em>Algorithms for
Molecular Biology</em>, <strong>14</strong>, 22.</p>
<p>[2] Dixon J.R., <em>et al</em> (2012). Topological domains in
mammalian genomes identified by analysis of chromatin interactions.
<em>Nature</em>, <strong>485</strong>(7398), 376.</p>
<p>[3] Randriamihamison N., Vialaneix N., and Neuvial P. (2021).
Applicability and interpretability of Wardâ€™s hierarchical agglomerative
clustering with or without contiguity constraints. <em>Journal of
Classification</em>, <strong>38</strong>, 363â€“389.</p>
<p>[4] Servant N., <em>et al</em> (2012). HiTC: Exploration of
High-Throughput â€˜Câ€™ experiments. <em>Bioinformatics</em>,
<strong>28</strong>(21), 2843-2844.</p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.5 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] adjclust_0.6.10      HiTC_1.48.0          GenomicRanges_1.56.1</span></span>
<span><span class="co">## [4] GenomeInfoDb_1.40.1  IRanges_2.38.1       S4Vectors_0.42.1    </span></span>
<span><span class="co">## [7] BiocGenerics_0.50.0 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] SummarizedExperiment_1.34.0 capushe_1.1.2              </span></span>
<span><span class="co">##  [3] gtable_0.3.5                rjson_0.2.23               </span></span>
<span><span class="co">##  [5] xfun_0.48                   bslib_0.8.0                </span></span>
<span><span class="co">##  [7] ggplot2_3.5.1               Biobase_2.64.0             </span></span>
<span><span class="co">##  [9] lattice_0.22-6              vctrs_0.6.5                </span></span>
<span><span class="co">## [11] tools_4.4.1                 bitops_1.0-9               </span></span>
<span><span class="co">## [13] curl_5.2.3                  parallel_4.4.1             </span></span>
<span><span class="co">## [15] fansi_1.0.6                 tibble_3.2.1               </span></span>
<span><span class="co">## [17] highr_0.11                  pkgconfig_2.0.3            </span></span>
<span><span class="co">## [19] Matrix_1.7-0                RColorBrewer_1.1-3         </span></span>
<span><span class="co">## [21] sparseMatrixStats_1.16.0    desc_1.4.3                 </span></span>
<span><span class="co">## [23] lifecycle_1.0.4             GenomeInfoDbData_1.2.12    </span></span>
<span><span class="co">## [25] compiler_4.4.1              Rsamtools_2.20.0           </span></span>
<span><span class="co">## [27] textshaping_0.4.0           Biostrings_2.72.1          </span></span>
<span><span class="co">## [29] munsell_0.5.1               codetools_0.2-20           </span></span>
<span><span class="co">## [31] htmltools_0.5.8.1           sass_0.4.9                 </span></span>
<span><span class="co">## [33] RCurl_1.98-1.16             yaml_2.3.10                </span></span>
<span><span class="co">## [35] pillar_1.9.0                pkgdown_2.1.1              </span></span>
<span><span class="co">## [37] crayon_1.5.3                jquerylib_0.1.4            </span></span>
<span><span class="co">## [39] MASS_7.3-60.2               BiocParallel_1.38.0        </span></span>
<span><span class="co">## [41] cachem_1.1.0                DelayedArray_0.30.1        </span></span>
<span><span class="co">## [43] viridis_0.6.5               abind_1.4-8                </span></span>
<span><span class="co">## [45] digest_0.6.37               restfulr_0.0.15            </span></span>
<span><span class="co">## [47] fastmap_1.2.0               grid_4.4.1                 </span></span>
<span><span class="co">## [49] colorspace_2.1-1            cli_3.6.3                  </span></span>
<span><span class="co">## [51] SparseArray_1.4.8           magrittr_2.0.3             </span></span>
<span><span class="co">## [53] S4Arrays_1.4.1              utf8_1.2.4                 </span></span>
<span><span class="co">## [55] XML_3.99-0.17               UCSC.utils_1.0.0           </span></span>
<span><span class="co">## [57] scales_1.3.0                rmarkdown_2.28             </span></span>
<span><span class="co">## [59] XVector_0.44.0              httr_1.4.7                 </span></span>
<span><span class="co">## [61] matrixStats_1.4.1           gridExtra_2.3              </span></span>
<span><span class="co">## [63] ragg_1.3.3                  evaluate_1.0.0             </span></span>
<span><span class="co">## [65] knitr_1.48                  BiocIO_1.14.0              </span></span>
<span><span class="co">## [67] viridisLite_0.4.2           rtracklayer_1.64.0         </span></span>
<span><span class="co">## [69] rlang_1.1.4                 dendextend_1.18.0          </span></span>
<span><span class="co">## [71] Rcpp_1.0.13                 glue_1.8.0                 </span></span>
<span><span class="co">## [73] jsonlite_1.8.9              R6_2.5.1                   </span></span>
<span><span class="co">## [75] MatrixGenerics_1.16.0       GenomicAlignments_1.40.0   </span></span>
<span><span class="co">## [77] systemfonts_1.1.0           fs_1.6.4                   </span></span>
<span><span class="co">## [79] zlibbioc_1.50.0</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="http://www.math-evry.cnrs.fr/members/cambroise/" class="external-link">Christophe Ambroise</a>, Shubham Chaturvedi, Alia Dehman, <a href="https://www.math.univ-toulouse.fr/~pneuvial/" class="external-link">Pierre Neuvial</a>, Guillem Rigaill, <a href="http://www.nathalievialaneix.eu" class="external-link">Nathalie Vialaneix</a>, Gabriel Hoffman.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
