<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="adjclust">
<title>Clustering of Hi-C contact maps • adjclust</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Clustering of Hi-C contact maps">
<meta property="og:description" content="adjclust">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">adjclust</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.6.7</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/hicClust.html">Clustering of Hi-C contact maps</a>
    <a class="dropdown-item" href="../articles/notesCHAC.html">Implementation notes for the `adjclust` package</a>
    <a class="dropdown-item" href="../articles/snpClust.html">Inferring Linkage Disequilibrium blocks from genotypes</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/pneuvial/adjclust/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Clustering of Hi-C contact maps</h1>
                        <h4 data-toc-skip class="author">Shubham
Chaturvedi, Pierre Neuvial, Nathalie Vialaneix</h4>
            
            <h4 data-toc-skip class="date">2023-04-28</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pneuvial/adjclust/blob/HEAD/vignettes/hicClust.Rmd" class="external-link"><code>vignettes/hicClust.Rmd</code></a></small>
      <div class="d-none name"><code>hicClust.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># IMPORTANT: this vignette can not be created if HiTC is not installed</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"HiTC"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>eval <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">##     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">##     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">##     Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,</span></span>
<span><span class="co">##     table, tapply, union, unique, unsplit, which.max, which.min</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'S4Vectors'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:utils':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     findMatches</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     expand.grid, I, unname</span></span></code></pre>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Hi-C is a sequencing-based molecular assay designed to measure intra
and inter-chromosomal interactions between the DNA molecule. In
particular, the identification of Topologically-Associated Domains
(TADs), that is, of regions of the genome in which physical interactions
are frequent, provides insight into the three-dimensional organization
of a genome [2].</p>
<p>Hi-C data are in the form of two-dimensional <em>contact maps</em>,
<em>i.e.</em>, matrices whose <span class="math inline">\(i,j\)</span>
entry quantifies the intensity of the physical interaction between two
genome regions <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> at the DNA level. In this vignette, we
demonstrate the use of <code><a href="../reference/hicClust.html">adjclust::hicClust</a></code> to perform
adjacency-constrained hierarchical agglomerative clustering (HAC) of
Hi-C contact maps. The output of this function is a dendrogram, which
can be cut to identify TADs. The algorithm used for
adjacency-constrained (HAC) is described in [3,4].</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://pneuvial.github.io/adjclust/">"adjclust"</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="loading-and-displaying-a-sample-hi-c-contact-map">Loading and displaying a sample Hi-C contact map<a class="anchor" aria-label="anchor" href="#loading-and-displaying-a-sample-hi-c-contact-map"></a>
</h2>
<p>The data set <code>hic_imr90_40_XX</code> is an object of class
<code>HTCexp</code> which has been obtained from the <code>HiTC</code>
package [4]. It is a contact map corresponding to the first 500 x 500
bins on chromosome X vs chromosome X.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"hic_imr90_40_XX.rda"</span>, package <span class="op">=</span> <span class="st">"adjclust"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>The script used to create this map can be found by executing the
following command:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"system/create_hic_chrXchrX.R"</span>, package<span class="op">=</span><span class="st">"adjclust"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "/home/runner/work/_temp/Library/adjclust/system/create_hic_chrXchrX.R"</span></span></code></pre>
<p>Now we have a look at the data.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">HiTC</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiTC/man/mapC.html" class="external-link">mapC</a></span><span class="op">(</span><span class="va">hic_imr90_40_XX</span><span class="op">)</span></span></code></pre></div>
<p><img src="hicClust_files/figure-html/mapHiC-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="using-hicclust">Using <code>hicClust</code><a class="anchor" aria-label="anchor" href="#using-hicclust"></a>
</h2>
<p><code>hicClust</code> operates directly on objects of class
<code>HTCexp</code></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hicClust.html">hicClust</a></span><span class="op">(</span><span class="va">hic_imr90_40_XX</span><span class="op">)</span></span></code></pre></div>
<p>It is also possible to work on binned data. Below we choose a bin
size of <span class="math inline">\(5 \times 10^5\)</span>:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">binned</span> <span class="op">&lt;-</span> <span class="fu">HiTC</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiTC/man/binningC.html" class="external-link">binningC</a></span><span class="op">(</span><span class="va">hic_imr90_40_XX</span>, binsize <span class="op">=</span> <span class="fl">1e5</span><span class="op">)</span></span>
<span><span class="va">fitB</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hicClust.html">hicClust</a></span><span class="op">(</span><span class="va">binned</span><span class="op">)</span></span>
<span><span class="va">fitB</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Call:</span></span>
<span><span class="co">## hicClust(binned)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Cluster method   : hicClust </span></span>
<span><span class="co">## Number of objects: 205</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">HiTC</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiTC/man/mapC.html" class="external-link">mapC</a></span><span class="op">(</span><span class="va">binned</span><span class="op">)</span></span></code></pre></div>
<p><img src="hicClust_files/figure-html/plotBinned-1.png" width="700"></p>
<p>The output is of class <code>chac</code>. In particular, it can be
plotted as a dendrogram silently relying on the function
<code>plot.dendrogram</code>:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fitB</span>, mode <span class="op">=</span> <span class="st">"corrected"</span><span class="op">)</span></span></code></pre></div>
<p><img src="hicClust_files/figure-html/dendro-1.png" width="700"></p>
<p>Moreover, the output contains an element named <code>merge</code>
which describes the successive merges of the clustering, and an element
<code>gains</code> which gives the improvement in the criterion
optimized by the clustering at each successive merge.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">fitB</span><span class="op">$</span><span class="va">merge</span>, <span class="va">fitB</span><span class="op">$</span><span class="va">gains</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##      [,1] [,2]</span></span>
<span><span class="co">## [1,]   -3   -4</span></span>
<span><span class="co">## [2,]   -2    1</span></span>
<span><span class="co">## [3,]    2   -5</span></span>
<span><span class="co">## [4,]   -1    3</span></span>
<span><span class="co">## [5,]    4   -6</span></span>
<span><span class="co">## [6,]  -17  -18</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="other-types-of-input">Other types of input<a class="anchor" aria-label="anchor" href="#other-types-of-input"></a>
</h2>
<p>Contacts maps can also be stored as objects of class
<code>Matrix::dsCMatrix</code>, or as plain text files. These types of
input are also accepted as first argument to <code>hicClust</code>.</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>[1] Ambroise C., Dehman A., Neuvial P., Rigaill G., and Vialaneix N.
(2019). Adjacency-constrained hierarchical clustering of a band
similarity matrix with application to genomics. <em>Algorithms for
Molecular Biology</em>, <strong>14</strong>, 22.</p>
<p>[2] Dixon J.R., <em>et al</em> (2012). Topological domains in
mammalian genomes identified by analysis of chromatin interactions.
<em>Nature</em>, <strong>485</strong>(7398), 376.</p>
<p>[3] Randriamihamison N., Vialaneix N., and Neuvial P. (2021).
Applicability and interpretability of Ward’s hierarchical agglomerative
clustering with or without contiguity constraints. <em>Journal of
Classification</em>, <strong>38</strong>, 363–389.</p>
<p>[4] Servant N., <em>et al</em> (2012). HiTC: Exploration of
High-Throughput ‘C’ experiments. <em>Bioinformatics</em>,
<strong>28</strong>(21), 2843-2844.</p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] adjclust_0.6.7       HiTC_1.44.0          GenomicRanges_1.52.0</span></span>
<span><span class="co">## [4] GenomeInfoDb_1.36.0  IRanges_2.34.0       S4Vectors_0.38.0    </span></span>
<span><span class="co">## [7] BiocGenerics_0.46.0 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] SummarizedExperiment_1.30.0 capushe_1.1.1              </span></span>
<span><span class="co">##  [3] gtable_0.3.3                rjson_0.2.21               </span></span>
<span><span class="co">##  [5] xfun_0.39                   bslib_0.4.2                </span></span>
<span><span class="co">##  [7] ggplot2_3.4.2               Biobase_2.60.0             </span></span>
<span><span class="co">##  [9] lattice_0.21-8              vctrs_0.6.2                </span></span>
<span><span class="co">## [11] tools_4.3.0                 bitops_1.0-7               </span></span>
<span><span class="co">## [13] parallel_4.3.0              fansi_1.0.4                </span></span>
<span><span class="co">## [15] tibble_3.2.1                highr_0.10                 </span></span>
<span><span class="co">## [17] pkgconfig_2.0.3             Matrix_1.5-4               </span></span>
<span><span class="co">## [19] RColorBrewer_1.1-3          sparseMatrixStats_1.12.0   </span></span>
<span><span class="co">## [21] desc_1.4.2                  lifecycle_1.0.3            </span></span>
<span><span class="co">## [23] GenomeInfoDbData_1.2.10     compiler_4.3.0             </span></span>
<span><span class="co">## [25] stringr_1.5.0               Rsamtools_2.16.0           </span></span>
<span><span class="co">## [27] textshaping_0.3.6           Biostrings_2.68.0          </span></span>
<span><span class="co">## [29] munsell_0.5.0               codetools_0.2-19           </span></span>
<span><span class="co">## [31] htmltools_0.5.5             sass_0.4.5                 </span></span>
<span><span class="co">## [33] RCurl_1.98-1.12             yaml_2.3.7                 </span></span>
<span><span class="co">## [35] pillar_1.9.0                pkgdown_2.0.7              </span></span>
<span><span class="co">## [37] crayon_1.5.2                jquerylib_0.1.4            </span></span>
<span><span class="co">## [39] MASS_7.3-58.4               BiocParallel_1.34.0        </span></span>
<span><span class="co">## [41] cachem_1.0.7                DelayedArray_0.25.0        </span></span>
<span><span class="co">## [43] viridis_0.6.2               digest_0.6.31              </span></span>
<span><span class="co">## [45] stringi_1.7.12              purrr_1.0.1                </span></span>
<span><span class="co">## [47] restfulr_0.0.15             rprojroot_2.0.3            </span></span>
<span><span class="co">## [49] fastmap_1.1.1               grid_4.3.0                 </span></span>
<span><span class="co">## [51] colorspace_2.1-0            cli_3.6.1                  </span></span>
<span><span class="co">## [53] magrittr_2.0.3              utf8_1.2.3                 </span></span>
<span><span class="co">## [55] XML_3.99-0.14               scales_1.2.1               </span></span>
<span><span class="co">## [57] rmarkdown_2.21              XVector_0.40.0             </span></span>
<span><span class="co">## [59] matrixStats_0.63.0          gridExtra_2.3              </span></span>
<span><span class="co">## [61] ragg_1.2.5                  memoise_2.0.1              </span></span>
<span><span class="co">## [63] evaluate_0.20               knitr_1.42                 </span></span>
<span><span class="co">## [65] BiocIO_1.10.0               viridisLite_0.4.1          </span></span>
<span><span class="co">## [67] rtracklayer_1.60.0          rlang_1.1.0                </span></span>
<span><span class="co">## [69] dendextend_1.17.1           Rcpp_1.0.10                </span></span>
<span><span class="co">## [71] glue_1.6.2                  jsonlite_1.8.4             </span></span>
<span><span class="co">## [73] R6_2.5.1                    MatrixGenerics_1.12.0      </span></span>
<span><span class="co">## [75] GenomicAlignments_1.36.0    systemfonts_1.0.4          </span></span>
<span><span class="co">## [77] fs_1.6.2                    zlibbioc_1.46.0</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="http://www.math-evry.cnrs.fr/members/cambroise/" class="external-link">Christophe Ambroise</a>, Shubham Chaturvedi, Alia Dehman, <a href="https://www.math.univ-toulouse.fr/~pneuvial/" class="external-link">Pierre Neuvial</a>, Guillem Rigaill, <a href="http://www.nathalievialaneix.eu" class="external-link">Nathalie Vialaneix</a>, Gabriel Hoffman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
