<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Inferring Linkage Disequilibrium blocks from genotypes • adjclust</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Inferring Linkage Disequilibrium blocks from genotypes">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">adjclust</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.6.10</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/hicClust.html">Clustering of Hi-C contact maps</a></li>
    <li><a class="dropdown-item" href="../articles/notesCHAC.html">Implementation notes for the `adjclust` package</a></li>
    <li><a class="dropdown-item" href="../articles/snpClust.html">Inferring Linkage Disequilibrium blocks from genotypes</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/pneuvial/adjclust/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Inferring Linkage Disequilibrium blocks from genotypes</h1>
                        <h4 data-toc-skip class="author">Shubham
Chaturvedi, Pierre Neuvial, Nathalie Vialaneix</h4>
            
            <h4 data-toc-skip class="date">2024-10-08</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pneuvial/adjclust/blob/master/vignettes/snpClust.Rmd" class="external-link"><code>vignettes/snpClust.Rmd</code></a></small>
      <div class="d-none name"><code>snpClust.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># IMPORTANT: this vignette is not created if snpStats is not installed</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"snpStats"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>eval <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: snpStats</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: survival</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Matrix</span></span></code></pre>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>In this vignette we demonstrate the use of <code>snpClust</code>
function in the <code>adjclust</code> package. <code>snpClust</code>
performs adjacency-constrained hierarchical clustering of single
nucleotide polymorphisms (SNPs), where the similarity between SNPs is
defined by linkage disequilibrium (LD).</p>
<p>This function implements the algorithm described in [1]. It is an
extension of the algorithm described in [3,4]. Denoting by
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>p</mi><annotation encoding="application/x-tex">p</annotation></semantics></math>
the number of SNPs to cluster and assuming that the similarity between
SNPs whose indices are more distant than
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>h</mi><annotation encoding="application/x-tex">h</annotation></semantics></math>,
its time complexity is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>O</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mo>log</mo><mrow><mo stretchy="true" form="prefix">(</mo><mi>p</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><mi>h</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">O(p (\log(p) + h))</annotation></semantics></math>,
and its space complexity is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>O</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>h</mi><mi>p</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">O(hp)</annotation></semantics></math>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://pneuvial.github.io/adjclust/">"adjclust"</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="loading-and-displaying-genotype-data">Loading and displaying genotype data<a class="anchor" aria-label="anchor" href="#loading-and-displaying-genotype-data"></a>
</h2>
<p>The beginning of this vignette closely follows the “LD vignette” of
the SnpStats package [2]. First, we load genotype data.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"ld.example"</span>, package <span class="op">=</span> <span class="st">"snpStats"</span><span class="op">)</span></span></code></pre></div>
<p>We focus on the <code>ceph.1mb</code> data.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geno</span> <span class="op">&lt;-</span> <span class="va">ceph.1mb</span><span class="op">[</span>, <span class="op">-</span><span class="fl">316</span><span class="op">]</span>  <span class="co">## drop one SNP leading to one missing LD value</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">geno</span><span class="op">)</span></span>
<span><span class="va">nSamples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">geno</span><span class="op">)</span></span>
<span><span class="va">geno</span></span></code></pre></div>
<pre><code><span><span class="co">## A SnpMatrix with  90 rows and  602 columns</span></span>
<span><span class="co">## Row names:  NA06985 ... NA12892 </span></span>
<span><span class="co">## Col names:  rs5993821 ... rs5747302</span></span></code></pre>
<p>These data are drawn from the International HapMap Project and
concern 602 SNPs<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;We have dropped SNP rs2401075 because it produced a
missing value due to the lack of genetic diversity in the considered
sample.&lt;/p&gt;"><sup>1</sup></a> over a 1Mb region of chromosome 22 in
sample of 90 Europeans.</p>
<p>We can compute and display the LD between these SNPs.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ld.ceph</span> <span class="op">&lt;-</span> <span class="fu">snpStats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/snpStats/man/ld.html" class="external-link">ld</a></span><span class="op">(</span><span class="va">geno</span>, stats <span class="op">=</span> <span class="st">"R.squared"</span>, depth <span class="op">=</span> <span class="va">p</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/image.html" class="external-link">image</a></span><span class="op">(</span><span class="va">ld.ceph</span>, lwd <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<p><img src="snpClust_files/figure-html/LD-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="adjacency-constrained-hierarchical-agglomerative-clustering">Adjacency-constrained Hierarchical Agglomerative Clustering<a class="anchor" aria-label="anchor" href="#adjacency-constrained-hierarchical-agglomerative-clustering"></a>
</h2>
<p>The <code>snpClust</code> function can handle genotype data as an
input:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/snpClust.html">snpClust</a></span><span class="op">(</span><span class="va">geno</span>, stats <span class="op">=</span> <span class="st">"R.squared"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in run.snpClust(x, h = h, stats = stats): Forcing the LD similarity to</span></span>
<span><span class="co">## be smaller than or equal to 1</span></span></code></pre>
<pre><code><span><span class="co">## Note: 135 merges with non increasing heights.</span></span></code></pre>
<p>Note that due to numerical errors in the LD estimation, some of the
estimated LD values may be slightly larger than 1. These values are
rounded to 1 internally.</p>
<p>The above figure suggests that the LD signal is concentrated close to
the diagonal. We can focus on a diagonal band with the bandwidth
parameter <code>h</code>:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fitH</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/snpClust.html">snpClust</a></span><span class="op">(</span><span class="va">geno</span>, h <span class="op">=</span> <span class="fl">100</span>, stats <span class="op">=</span> <span class="st">"R.squared"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in run.snpClust(x, h = h, stats = stats): Forcing the LD similarity to</span></span>
<span><span class="co">## be smaller than or equal to 1</span></span></code></pre>
<pre><code><span><span class="co">## Note: 133 merges with non increasing heights.</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fitH</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Call:</span></span>
<span><span class="co">## snpClust(geno, h = 100, stats = "R.squared")</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Cluster method   : snpClust </span></span>
<span><span class="co">## Number of objects: 602</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="output">Output<a class="anchor" aria-label="anchor" href="#output"></a>
</h2>
<p>The output of the <code>snpClust</code> is of class
<code>chac</code>. In particular, it can be plotted as a dendrogram
silently relying on the function <code>plot.dendrogram</code>:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fitH</span>, type <span class="op">=</span> <span class="st">"rectangle"</span>, leaflab <span class="op">=</span> <span class="st">"perpendicular"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: </span></span>
<span><span class="co">## Detected reversals in dendrogram: mode = 'corrected', 'within-disp' or 'total-disp' might be more relevant.</span></span></code></pre>
<p><img src="snpClust_files/figure-html/dendro-1.png" width="700"></p>
<p>Moreover, the output contains an element named <code>merge</code>
which describes the successive merges of the clustering, and an element
<code>gains</code> which gives the improvement in the criterion
optimized by the clustering at each successive merge.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">fitH</span><span class="op">$</span><span class="va">merge</span>, <span class="va">fitH</span><span class="op">$</span><span class="va">gains</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##      [,1] [,2]</span></span>
<span><span class="co">## [1,]   -1   -2</span></span>
<span><span class="co">## [2,] -255 -256</span></span>
<span><span class="co">## [3,] -488 -489</span></span>
<span><span class="co">## [4,] -487    3</span></span>
<span><span class="co">## [5,] -486    4</span></span>
<span><span class="co">## [6,] -234 -235</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="other-types-of-input">Other types of input<a class="anchor" aria-label="anchor" href="#other-types-of-input"></a>
</h2>
<p>In this section we show how the <code>snpClust</code> function can
also be applied directly to LD values.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h</span> <span class="op">&lt;-</span> <span class="fl">100</span></span>
<span><span class="va">ld.ceph</span> <span class="op">&lt;-</span> <span class="fu">snpStats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/snpStats/man/ld.html" class="external-link">ld</a></span><span class="op">(</span><span class="va">geno</span>, stats <span class="op">=</span> <span class="st">"R.squared"</span>, depth <span class="op">=</span> <span class="va">h</span>, symmetric <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/image.html" class="external-link">image</a></span><span class="op">(</span><span class="va">ld.ceph</span>, lwd <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<p><img src="snpClust_files/figure-html/snpClust-LD-1.png" width="700"></p>
<p>Note that we have forced the <code><a href="https://rdrr.io/pkg/snpStats/man/ld.html" class="external-link">snpStats::ld</a></code> function to
return a symmetric matrix. We can apply <code>snpClust</code> directly
to this LD matrix (of class <code>Matrix::dsCMatrix</code>):</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fitL</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/snpClust.html">snpClust</a></span><span class="op">(</span><span class="va">ld.ceph</span>, <span class="va">h</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Note: forcing the diagonal of the LD similarity matrix to be 1</span></span></code></pre>
<pre><code><span><span class="co">## Warning in run.snpClust(x, h = h, stats = stats): Forcing the LD similarity to</span></span>
<span><span class="co">## be smaller than or equal to 1</span></span></code></pre>
<pre><code><span><span class="co">## Note: 133 merges with non increasing heights.</span></span></code></pre>
<p><code>snpClust</code> also handles inputs of class
<code><a href="https://rdrr.io/r/base/matrix.html" class="external-link">base::matrix</a></code>:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">geno</span>, <span class="st">"matrix"</span><span class="op">)</span></span>
<span><span class="va">fitM</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/snpClust.html">snpClust</a></span><span class="op">(</span><span class="va">geno</span>, <span class="va">h</span>, stats <span class="op">=</span> <span class="st">"R.squared"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Note: 133 merges with non increasing heights.</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>[1] Ambroise C., Dehman A., Neuvial P., Rigaill G., and Vialaneix N.
(2019). Adjacency-constrained hierarchical clustering of a band
similarity matrix with application to genomics. <em>Algorithms for
Molecular Biology</em>, <strong>14</strong>, 22.</p>
<p>[2] Clayton D. (2015). snpStats: SnpMatrix and XSnpMatrix classes and
methods. R package version 1.20.0</p>
<p>[3] Dehman A., Ambroise C., Neuvial P. (2015). Performance of a
blockwise approach in variable selection using linkage disequilibrium
information. <em>BMC Bioinformatics</em>, <strong>16</strong>, 148.</p>
<p>[4] Randriamihamison N., Vialaneix N., and Neuvial P. (2021).
Applicability and interpretability of Ward’s hierarchical agglomerative
clustering with or without contiguity constraints. <em>Journal of
Classification</em>, <strong>38</strong>, 363–389.</p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.5 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] adjclust_0.6.10 snpStats_1.54.0 Matrix_1.7-0    survival_3.6-4 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] viridis_0.6.5            sass_0.4.9               utf8_1.2.4              </span></span>
<span><span class="co">##  [4] lattice_0.22-6           digest_0.6.37            magrittr_2.0.3          </span></span>
<span><span class="co">##  [7] evaluate_1.0.0           sparseMatrixStats_1.16.0 grid_4.4.1              </span></span>
<span><span class="co">## [10] fastmap_1.2.0            jsonlite_1.8.9           gridExtra_2.3           </span></span>
<span><span class="co">## [13] fansi_1.0.6              capushe_1.1.2            viridisLite_0.4.2       </span></span>
<span><span class="co">## [16] scales_1.3.0             textshaping_0.4.0        jquerylib_0.1.4         </span></span>
<span><span class="co">## [19] cli_3.6.3                rlang_1.1.4              munsell_0.5.1           </span></span>
<span><span class="co">## [22] splines_4.4.1            cachem_1.1.0             yaml_2.3.10             </span></span>
<span><span class="co">## [25] tools_4.4.1              colorspace_2.1-1         ggplot2_3.5.1           </span></span>
<span><span class="co">## [28] BiocGenerics_0.50.0      vctrs_0.6.5              R6_2.5.1                </span></span>
<span><span class="co">## [31] matrixStats_1.4.1        lifecycle_1.0.4          zlibbioc_1.50.0         </span></span>
<span><span class="co">## [34] fs_1.6.4                 MASS_7.3-60.2            ragg_1.3.3              </span></span>
<span><span class="co">## [37] pkgconfig_2.0.3          desc_1.4.3               dendextend_1.18.0       </span></span>
<span><span class="co">## [40] pkgdown_2.1.1            pillar_1.9.0             bslib_0.8.0             </span></span>
<span><span class="co">## [43] gtable_0.3.5             glue_1.8.0               Rcpp_1.0.13             </span></span>
<span><span class="co">## [46] systemfonts_1.1.0        highr_0.11               xfun_0.48               </span></span>
<span><span class="co">## [49] tibble_3.2.1             MatrixGenerics_1.16.0    knitr_1.48              </span></span>
<span><span class="co">## [52] htmltools_0.5.8.1        rmarkdown_2.28           compiler_4.4.1</span></span></code></pre>
</div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="http://www.math-evry.cnrs.fr/members/cambroise/" class="external-link">Christophe Ambroise</a>, Shubham Chaturvedi, Alia Dehman, <a href="https://www.math.univ-toulouse.fr/~pneuvial/" class="external-link">Pierre Neuvial</a>, Guillem Rigaill, <a href="http://www.nathalievialaneix.eu" class="external-link">Nathalie Vialaneix</a>, Gabriel Hoffman.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
